pkg/specgenutil/volumes.go

```diff
@@ -723,3 +703,26 @@ func validChownFlag(flag string) (bool, error) {
 func unixPathClean(p string) string {
 	return path.Clean(p)
 }
+
+func addReadOnlyMounts(mounts []spec.Mount) []spec.Mount {
+	readonlyTmpfs := []string{"/tmp", "/var/tmp", "/run"}
+	options := []string{"rw", "rprivate", "nosuid", "nodev", "tmpcopyup"}
+	for _, dest := range readonlyTmpfs {
+		for _, m := range mounts {
+			if m.Destination == dest {
+				continue
+			}
+		}
+		mnt := spec.Mount{
+			Destination: dest,
+			Type:        define.TypeTmpfs,
+			Source:      "tmpfs",
```

Comment:
OMG I think this is it. This is the bug, or maybe it's somewhere else, but I'm kind of sure this is it.

Here is what is happening, in a very brief nutshell:
```console
$ rm -rf tmpfs   ! humor me okay
$ bin/podman create --pod new:pod1 --read-only --name foo quay.io/libpod/testimage:20221018 touch /tmp/oh-hi-there
3d77b20d33a5fe8c27cd7c0051434c922d432fef2f4c320f41795ea8fa407889
$ bin/podman kube generate pod1 -f /tmp/foo.yaml
...
$ bin/podman kube play --replace /tmp/foo.yaml
...blah blah
$ ls -l tmpfs
total 0
-rw-r--r--. 1 esm esm 0 Nov 30 13:14 oh-hi-there
drwxr-xr-t. 1 esm esm 0 Nov 30 13:14 secrets/
```
That is: podman is creating a subdirectory called "tmpfs" in the current directory. I do not think that is what is intended. I think the intention is to mount an actual tmpfs.

The reason it failed on my laptop is that I ran `hack/bats`, which first runs as root, creates the root-owned subdirectory `tmpfs`, which then the second-pass (rootless) test of course fails to write with EACCES.

Reply:
Yeah, this is it. If I change "tmpfs" to "sdfsdfsdf", recompile and run, I get a new subdirectory with that name. OK, I'm done with this. Time for a nap. My brain hurts.

Reply:
Ok that is a different bug, not related to this PR.  But podman generate kube is generating a bogus mount point for tmpfs mounts.